File: Microsoft.Access.Data/Form1.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Equin.ApplicationFramework;
using Microsoft.Access.Data.Contexts;
using Microsoft.Access.Data.LanguageExtensions;
using Microsoft.Access.Data.WorkingClasses;
using Microsoft.EntityFrameworkCore;

namespace Microsoft.Access.Data
{
    public partial class Form1 : Form
    {
        //https://github.com/waynebloss/BindingListView
        private BindingListView<CustomerEntity> _customersView;
        private BindingSource _customersBindingSource = new BindingSource();
        public Form1()
        {
            InitializeComponent();
            Shown += Form1_Shown;
        }

        private void Form1_Shown(object sender, EventArgs e)
        {
            dataGridView1.AutoGenerateColumns = false;

            using (var context = new NorthWindAccessContext())
            {
                var customerData = (from customer in context.Customers
                    join country in context.Countries on customer.CountryIdentifier equals country.CountryIdentifier
                    select new CustomerEntity
                    {
                        CustomerIdentifier = customer.CustomerIdentifier,
                        CustomerName = customer.CompanyName,
                        CountryIdentifier = customer.CountryIdentifier,
                        Country = country.Name
                    }).ToList();

                _customersView = new BindingListView<CustomerEntity>(customerData);
                _customersBindingSource.DataSource = _customersView;
                dataGridView1.DataSource = _customersBindingSource;
            }
        }

        private void CurrentButton_Click(object sender, EventArgs e)
        {
            var currentCustomer = _customersBindingSource.CurrentCustomerEntity();
            MessageBox.Show($"Id: {currentCustomer.CustomerIdentifier} country id: {currentCustomer.CountryIdentifier}");
        }
    }
}

File: Converted/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports WindowsFormsApp1.Models
Imports Microsoft.EntityFrameworkCore

Namespace WindowsFormsApp1
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			If Not Me.button1.IsHandleCreated Then Return

			Using context = New NorthContext()
				Dim results = context.Customers.Include(Function(item) item.ContactTitleNavigation).ToList()
				Console.WriteLine()
			End Using
		End Sub
	End Class
End Namespace
